### PR TITLE
Configure Rails annotations via environment config (Rails 6+)

### PR DESCRIPTION
**Pull request by orien**
### Summary

Fix `rails notes` support by adding config to `config/environments/development.rb` and `config/environments/test.rb` for Rails 6+ projects..

### Context

As per #510, the config generated via `rails generate cucumber:install` does not support `rails notes` for Rails 6+.

By [changing the annotate feature](https://github.com/cucumber/cucumber-rails/pull/511/commits/4e751e5042ad8d7137795d22d57ed3d32a8883f0) to exercise `rails notes` rather than `rake notes` the problem is [exposed](https://github.com/cucumber/cucumber-rails/runs/2284601677).

### Change

Update the install generator so when `rails generate cucumber:install` is run the existing Rake-based annotation configuration is only generated for projects running versions of Rails pre v6.0. When running Rails v6.0 or greater, add configuration to the Rails `development` and `test` environment configuration files.

#### How Has This Been Tested?

This is covered in both RSpec and Cucumber tests.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

### Checklist

- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: spec/generators/cucumber/install_generator_spec.rb [hunk 1]
```diff
@@ -7,7 +7,16 @@
...
-  before { prepare_destination }
+  before do
+    prepare_destination
+    FileUtils.mkdir_p File.join(destination_root, 'config', 'environments')
+    %w[development test].each do |environment|
+      File.write(
+        File.join(destination_root, 'config', 'environments', "#{environment}.rb"),
+        "Rails.application.configure do\nend"
+      )
+    end
```
Comment (orien) [inline spec/generators/cucumber/install_generator_spec.rb:18]: Given the generator is now modifying existing configuration files, we'll need to ensure these files exist before exercising the system-under-test in the RSpec test suite.

Comment (olleolleolle) [reply spec/generators/cucumber/install_generator_spec.rb:18]: Cool, perhaps those new lines could have their own local helper method, with a name that captures your contextual comment?

**Review by olleolleolle — COMMENTED** [review]
Super-nice addition!

File: spec/generators/cucumber/install_generator_spec.rb (hunk 1 above)
Comment (orien) [reply spec/generators/cucumber/install_generator_spec.rb:22]: Excellent idea! Added in [`eca5bad`](https://github.com/cucumber/cucumber-rails/compare/884b75e689f15b7e8e335b397f3e586c00276392..eca5bad975acca459a3bf629888cbb8932ca59e2).

**Comment by luke-hill** [comment]
This explains a lot (I was getting the rake notes test failing locally!)

**Pull request merged**